### PR TITLE
Fix position in category setter from WS

### PR DESCRIPTION
**Pull request by TomLorenzi**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | When trying to fetch and update a product using the WS you'll get this error if you fetch and return all the writable informations, the modified code seems obvious that if you prepend null and try to find a key on an null object it will throw an error
| Type?             | bug fix
| Category?         | WS
| BC breaks?        | idk what it means
| Deprecations?     | no
| How to test?      | Try to fetch a product with `'display' => 'full'`, remove `manufacturer_name` and `quantity` bc it's not writable and push that product, you'll get this 500 error as a NOTICE
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/31252
| Related PRs       | None
| Sponsor company   |

### CODE REVIEW

**Comment by ghost** [comment]
Thank you for your PR

Can you open an issue for the QA Team with the steps to reproduce. 

Thank you again

**Comment by TomLorenzi** [comment]
> Thank you for your PR
> 
> Can you open an issue for the QA Team with the steps to reproduce.
> 
> Thank you again

Done :)

File: classes/Product.php [hunk 1]
```diff
@@ -7176,7 +7176,9 @@ public function setWsPositionInCategory($position)
         // result is indexed by recordset order and not position. positions start at index 1 so we need an empty element
         array_unshift($result, null);
         foreach ($result as &$value) {
-            $value = $value['id_product'];
```
Comment (matthieu-rolland) [inline classes/Product.php:7179]: I don't understand why `$value` can be null sometimes ? Isn't that the bug that we should fix ? (while your php check is still a good addition)

**Review by matthieu-rolland — COMMENTED** [review]
I have a question as I'm not sure I understand what's the real issue here

File: classes/Product.php (hunk 1 above)
Comment (TomLorenzi) [reply classes/Product.php:7179]: `array_unshift($result, null);` Here is why the value is null the line just above.

Preprending the array with a null value and right after trying to access a key on a null object.

Maybe if it's really needed we should move that array_unshift after that foreach ?

Comment (matthieu-rolland) [reply classes/Product.php:7179]: ah yes I missed the array_unshift...

yes it seems like it doesn't need to be done before the foreach, moving it just after would fix the bug without adding more logic :+1:

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by TomLorenzi** [comment]
Any update on this ? @matthieu-rolland

**Comment by kpodemski** [comment]
Hi @TomLorenzi 

For some reason, you merged dev to your dev branch :) It would be best to start from a branch derived from the upstream dev branch. When you need to get some changes from dev to your branch, you can git pull with the rebase not to create merge commits.

**Review by kpodemski — CHANGES_REQUESTED** [review]
We don't accept merge commits. You have to [rebase](https://build.prestashop-project.org/howtos/misc/rebasing-your-git-branch/).

**Comment by TomLorenzi** [comment]
@kpodemski I think it might be better, I'm not sure if I've done it the right way ?

**Comment by TomLorenzi** [comment]
Guess the PR will get stuck with the "waiting for author" status ?

**Comment by matks** [comment]
@TomLorenzi we have detected that "Waiting for author" status sometimes wrongly stays so we are working on a fix ;) see https://github.com/PrestaShop/issuebot/pull/104

**Review by kpodemski — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Comment by AureRita** [comment]
Hi @TomLorenzi, 

Currently, your issue is not verified, we try to reproduce it with @boherm, we can't do this with the provided information, 

maybe if you provide a full video of your issue or the entire request ( with the body ) you sended on the WS, 
we could reproduce it. 

Until we didn't have this information, I'll let this PR at blocked 

Waiting for your feedback

**Comment by TomLorenzi** [comment]
Obvious watching the code that this will trigger a notice...
I'll try providing a video later

**Comment by TomLorenzi** [comment]
I don't have the environnement setup + video capturing tools anymore.
I won't be able to provide you a full video.

Here's the PR with an obvious error that will trigger a notice.
If you can't reproduce it then don't merge it then, but wouldn't be able to spend more time on this.

While btw i never got any answer here: https://github.com/PrestaShop/PrestaShop/issues/31252 trying to find why and what could cause that this is not reproduced.

**Comment by TomLorenzi** [comment]
Guess it will never be merged as this error is not obvious enought, can be closed then

**Comment by kpodemski** [comment]
Hello @TomLorenzi 

I reproduced the error, and your fix did help. I'll merge it as soon as the CI is green :)

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
CI is green, I've reproduced the issue and tested the PR myself.

**Comment by kpodemski** [comment]
thank you @TomLorenzi

**Pull request merged**